### PR TITLE
ARROW-738: Fix manylinux1 build

### PR DESCRIPTION
**Pull request by xhochy**

### CODE REVIEW

File: python/CMakeLists.txt [hunk 1]
```diff
@@ -226,24 +226,27 @@ endif()
...
+  SET(ARROW_SHARED_LIB
+      ${BUILD_OUTPUT_ROOT_DIRECTORY}/${LIBRARY_NAME}${CMAKE_SHARED_LIBRARY_SUFFIX})
+endfunction(bundle_arrow_lib)
+
+if (PYARROW_BUNDLE_ARROW_CPP)
+  # arrow
+  bundle_arrow_lib(ARROW_SHARED_LIB)
+  bundle_arrow_lib(ARROW_IO_SHARED_LIB)
+  bundle_arrow_lib(ARROW_IPC_SHARED_LIB)
+  bundle_arrow_lib(ARROW_PYTHON_SHARED_LIB)
```
Comment (wesm) [inline python/CMakeLists.txt]: nice

File: python/setup.py [hunk 2]
```diff
@@ -208,8 +208,12 @@ def _run_cmake(self):
...
             lib_filename = (shared_library_prefix + lib_name +
                             shared_library_suffix)
-            shutil.move(pjoin(self.build_type, lib_filename),
-                        pjoin(build_lib, 'pyarrow', lib_filename))
+            # Also copy libraries with ABI/SO version suffix
+            libs = glob.glob(pjoin(self.build_type, lib_filename) + '*')
+            for lib in libs:
+                filename = os.path.basename(lib)
+                shutil.move(pjoin(self.build_type, filename),
+                            pjoin(build_lib, 'pyarrow', filename))
```
Comment (wesm) [inline python/setup.py]: Some of these are symlinks, though, right? is there a way to only move the non-symlinked libs?

**Comment by wesm** [comment]
I had to add `import glob` to get this to work locally, also per the symlink thing I have

```
-rw-r--r-- 1 wesm wesm  1008788 Mar 31 15:03 libarrow_io.so
-rw-r--r-- 1 wesm wesm  1008788 Mar 31 15:06 libarrow_io.so.0
-rw-r--r-- 1 wesm wesm  1008788 Mar 31 15:06 libarrow_io.so.0.0.0
-rw-r--r-- 1 wesm wesm  5040094 Mar 31 15:03 libarrow_ipc.so
-rw-r--r-- 1 wesm wesm  5040094 Mar 31 15:06 libarrow_ipc.so.0
-rw-r--r-- 1 wesm wesm  5040094 Mar 31 15:06 libarrow_ipc.so.0.0.0
-rw-r--r-- 1 wesm wesm  3869440 Mar 31 15:03 libarrow_python.so
-rw-r--r-- 1 wesm wesm  3869440 Mar 31 15:06 libarrow_python.so.0
-rw-r--r-- 1 wesm wesm  3869440 Mar 31 15:06 libarrow_python.so.0.0.0
-rw-r--r-- 1 wesm wesm  7499685 Mar 31 15:03 libarrow.so
-rw-r--r-- 1 wesm wesm  7499685 Mar 31 15:06 libarrow.so.0
-rw-r--r-- 1 wesm wesm  7499685 Mar 31 15:06 libarrow.so.0.0.0
-rw-r--r-- 1 wesm wesm  4671965 Mar 31 15:03 libparquet_arrow.so
-rw-r--r-- 1 wesm wesm 15179907 Mar 31 15:03 libparquet.so
```

File: python/setup.py (hunk 2 above)
Comment (xhochy) [inline python/setup.py]: Done.

**Comment by wesm** [comment]
Cool. When we add Windows support for pyarrow (ARROW-734) we'll probably want to have a dedicated code path for non-Unix-y platforms

**Comment by wesm** [comment]
Looks like S3 is cranky today. Going to check this out locally and then merge

**Comment by wesm** [comment]
+1, looks good here. thanks @xhochy, and sorry about breaking the build!

**Pull request closed without merging**